File: frontend/src/types/wallet.ts
export interface WalletBalance {
  account_type: string;  // FUNDING or TRADING
  currency: string;      // USDT, USDC, BTC, USD, HKD, EUR
  balance: string;       // decimal string
  frozen_balance: string;
  available_balance: string;
}

export interface WalletBalancesResponse {
  wallets: WalletBalance[];
}

export type WalletLedgerEntryType = "CREDIT" | "FREEZE" | "UNFREEZE" | "DEDUCT_FROZEN";

export interface WalletLedgerEntry {
  id: number;
  account_type: string;
  currency: string;
  entry_type: WalletLedgerEntryType | string;
  amount: string;
  balance_before: string;
  balance_after: string;
  frozen_before: string;
  frozen_after: string;
  transaction_record_id?: number;
  platform_order_id?: string;
  biz_type?: string;
  remark?: string;
  created_at: string;
}

export interface WalletLedgerListResponse {
  entries: WalletLedgerEntry[];
  total: number;
  page: number;
  page_size: number;
}

export interface WalletLedgerQuery {
  page?: number;
  page_size?: number;
  account_type?: string;
  currency?: string;
  biz_type?: string;
  entry_type?: string;
}

File: backend/internal/dto/response/wallet.go
package response

type WalletBalanceResp struct {
	AccountType      string `json:"account_type"`
	Currency         string `json:"currency"`
	Balance          string `json:"balance"`
	FrozenBalance    string `json:"frozen_balance"`
	AvailableBalance string `json:"available_balance"`
}

type WalletBalancesResp struct {
	Wallets []WalletBalanceResp `json:"wallets"`
}

type WalletLedgerEntryResp struct {
	ID                  uint64  `json:"id"`
	AccountType         string  `json:"account_type"`
	Currency            string  `json:"currency"`
	EntryType           string  `json:"entry_type"`
	Amount              string  `json:"amount"`
	BalanceBefore       string  `json:"balance_before"`
	BalanceAfter        string  `json:"balance_after"`
	FrozenBefore        string  `json:"frozen_before"`
	FrozenAfter         string  `json:"frozen_after"`
	TransactionRecordID *uint64 `json:"transaction_record_id,omitempty"`
	PlatformOrderID     *string `json:"platform_order_id,omitempty"`
	BizType             *string `json:"biz_type,omitempty"`
	Remark              *string `json:"remark,omitempty"`
	CreatedAt           string  `json:"created_at"`
}

type WalletLedgerListResp struct {
	Entries  []WalletLedgerEntryResp `json:"entries"`
	Total    int64                   `json:"total"`
	Page     int                     `json:"page"`
	PageSize int                     `json:"page_size"`
}
